driver-sync/src/main/com/mongodb/client/ListCollectionsIterable.java

```diff
@@ -38,6 +39,21 @@
      */
     ListCollectionsIterable<TResult> filter(@Nullable Bson filter);
 
+    /**
+     * Sets the {@code authorizedCollections} field of the {@code listCollections} command.
+     * This method is ignored if called on a {@link ListCollectionsIterable} obtained not via any of the
+     * {@link MongoDatabase#listCollectionNames() MongoDatabase.listCollectionNames} methods.
+     *
+     * @param authorizedCollections If {@code true}, allows executing the {@code listCollections} command,
+     * which has the {@code nameOnly} field set to {@code true}, without having the
+     * <a href="https://docs.mongodb.com/manual/reference/privilege-actions/#mongodb-authaction-listCollections">
+     * {@code listCollections} privilege</a> on the corresponding database resource.
+     * @return {@code this}.
+     * @since 4.5
+     * @mongodb.server.release 4.0
+     */
+    ListCollectionsIterable<TResult> authorizedCollections(boolean authorizedCollections);
```

Comment:
This is a breaking change for implementors of the interface. It's possible to make this method a `default` one with the default implementation that either does nothing or throws an `UnsupportedOperationException`. However, given the following

- We discussed a similar approach in the context of a different change (I don't remember what it was), and no one was happy about either of the work-arounds.
- The changes in `com.mongodb.client.MongoDatabase` and `com.mongodb.reactivestreams.client.MongoDatabase` are also breaking for implementors, and it is impossible to make them backward-compatible.

I decided not to apply such work-arounds.

Reply:
We haven't considered these to be breaking changes in the past, and frequently make them in minor releases like 4.5.